index.js

```diff
@@ -18,37 +17,55 @@ function Plugin(/* config.port */karmaPort, /* config.hostname */hostname, /* co
 	this.waiting = [];
 	this.fileList = fileList;
 
-	webpackOptions.output.path = "/";
-	webpackOptions.output.filename = "_js/[name]";
-	webpackOptions.output.chunkFilename = "_js/[id].chunk.js";
-	webpackOptions.output.publicPath = "http://" + hostname + ":" + port + "/";
+	var applyOptions = Array.isArray(webpackOptions) ? webpackOptions : [webpackOptions];
+	applyOptions.forEach(function(webpackOptions) {
+		if(!webpackOptions.output) webpackOptions.output = {};
+		webpackOptions.output.path = "/";
+		webpackOptions.output.filename = "_js/[name]";
+		webpackOptions.output.chunkFilename = "_js/[id].chunk.js";
+		webpackOptions.output.publicPath = "http://" + hostname + ":" + port + "/";
+	});
 
 	var compiler = webpack(webpackOptions);
-	compiler.plugin("compilation", function(compilation, params) {
-		compilation.dependencyFactories.set(SingleEntryDependency, params.normalModuleFactory);
-	});
-	compiler.plugin("done", function(stats) {
-		var compilation = stats.compilation;
-		stats = stats.toJson();
-
-		if(!this.waiting || this.waiting.length === 0) {
-			// If file required in tests is changed, webpack compilation is done silently for karma.
-			// Fix this by emulating test file change.
-			this.notifyKarmaAboutChanges(stats);
-		}
-
-		if(this.waiting && stats.assets.length > 0) {
-			var w = this.waiting;
-			this.waiting = null;
-			w.forEach(function(cb) {
-				cb();
-			});
-		}
-	}.bind(this));
-	compiler.plugin("invalid", function() {
-		if(!this.waiting) this.waiting = [];
-	}.bind(this));
-	compiler.plugin("make", this.make.bind(this));
+	var applyPlugins = compiler.compilers || [compiler];
+	applyPlugins.forEach(function(compiler) {
+		compiler.plugin("compilation", function(compilation, params) {
```

Comment:
Some loader start child compilers. i. e. the worker-loader starts a child compiler for each WebWorker bundle. The `compilation` hook also applies to child compilations, but the `this-compilation` hook doesn't. But this is a minor detail you cannot know.